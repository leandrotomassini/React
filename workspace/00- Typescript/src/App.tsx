import { Counter } from './components/Counter';
import { TimerPadre } from './components/TimerPadre';

import { Usuario } from './components/Usuario';

function App() {
  return (
    <>

      <h1 className="pt-3">React + Typescript</h1>

      <hr />

      <h2>Use State</h2>
      <Counter />
      <Usuario />

      <h2>useEffect - useRef</h2>
      <hr />

      <TimerPadre/>

    </>
  );
}

export default App;
