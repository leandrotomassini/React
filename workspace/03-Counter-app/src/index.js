import { render } from "@testing-library/react";
import CounterApp from "./CounterApp";


import './index.css';

const divRoot = document.querySelector('#root');

render(<CounterApp value={8}/>, divRoot);


