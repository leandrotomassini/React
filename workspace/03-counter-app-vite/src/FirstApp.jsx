import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {


    return (
        <>
            <h1>{title}</h1>
            <p>{subtitle}</p>
            <span>{name}</span>
        </>
    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
};


FirstApp.defaultProps = {
    name: 'Leandro',
    subtitle: 'No hay subtítulo',
    title: 'No hay título',
}