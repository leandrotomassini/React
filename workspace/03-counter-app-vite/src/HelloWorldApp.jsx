export const HelloWorldApp = () => {
    return (
        <h1>HelloWorldApp</h1>
    )
}
