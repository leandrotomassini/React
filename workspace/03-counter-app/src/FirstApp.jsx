import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {


  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <label>{name}</label>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
}

FirstApp.defaultProps = {
  name: 'Leandro C. Tomassini',
  subtitle: 'No hay subtítulo',
  title: 'No hay título'
};