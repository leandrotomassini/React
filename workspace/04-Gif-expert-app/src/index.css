* {
    font-family: Helvetica, Arial, sans-serif;
}

body {
    padding: 60px;
}

input {
    color: grey;
    font-size: 1.3rem;
    width: 100%;
}