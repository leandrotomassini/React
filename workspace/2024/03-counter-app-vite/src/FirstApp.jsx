import PropTypes from 'prop-types';


export const FirstApp = ({
    title,
    subtitle,
    name
}) => {



    return (
        <>
            <div data-testid="test-title">{title}</div>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
};

FirstApp.defaultProps = {
    title: "No hay título",
    subtitle: "No hay subtítulo"
};

