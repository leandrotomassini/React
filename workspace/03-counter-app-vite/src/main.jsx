<<<<<<< HEAD
import React from 'react';
import ReactDOM from 'react-dom/client';

import { CounterApp } from './CounterApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={10} />
    </React.StrictMode>
);
=======
import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App.jsx'
import './index.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
)
>>>>>>> c32a58255c4c3b65fbd5646b231cbe052c117e6f
