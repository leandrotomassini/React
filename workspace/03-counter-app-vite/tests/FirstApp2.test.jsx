import { render, screen } from "@testing-library/react";

import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {

    const title = 'Hola, soy Goku';
    const subTitle = 'Soy un subtítulo';

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje "Hola, soy Goku"', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debe de mostrar el título en un H1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML.toContain(title));
    });

    test('Debe de mostrar el subtítulo enviado por props', () => {
        render(<FirstApp title={title} subtitle={subTitle} />);
        expect(screen.getAllByText(subTitle).length).toBe(2);
    });
});