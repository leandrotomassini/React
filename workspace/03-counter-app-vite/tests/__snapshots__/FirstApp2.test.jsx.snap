// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Goku
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
